content/protobuf/language-mapping/service.md

```diff
@@ -0,0 +1,220 @@
+The `protoc-gen-icerpc-csharp` plugin compiles Protobuf service _Name_ into two C# interfaces (I*Name* and
+I*Name*Service) and one C# struct (*Name*Client). The identifiers of the generated interfaces and struct are always in
+Pascal case, per the usual C# naming conventions, even when _Name_ is not in Pascal case.
+
+### I*Name*
+
+Interface I*Name* provides the client-side API that allows you to call the remote IceRPC service that implements the
+the associated Protobuf service. It's a minimal interface with an abstract method for each RPC method defined in this
+Protobuf service.
+
+For example:
+
+{% side-by-side alignment="top" %}
+
+```protobuf
+syntax = "proto3";
+package example;
+option csharp_namespace = "Example";
+
+service Widget {
+    rpc Spin (SpinConfig)
+        returns (google.protobuf.Empty)
+}
+```
+
+```csharp
+namespace Example;
+
+public partial interface IWidget
+{
+    // One C# method per rpc method
+    Task<WellKnownTypes.Empty> SpinAsync(
+        SpinConfig message,
+        IFeatureCollection? features = null,
+        CancellationToken cancellationToken =
+            default);
+}
+```
+
+{% /side-by-side %}
+
+### *Name*Client
+
+The generated record struct *Name*Client implements I*Name* by sending requests to a remote service with IceRPC.
+
+In order to call a remote service, you need to construct a client struct using one of its "invoker" constructors:
+
+```csharp
+public readonly partial record struct WidgetProxy : IWidget
+{
+    public WidgetProxy(
+        IInvoker invoker,
+        ServiceAddress? serviceAddress = null,
+        ProtobufEncodeOptions? encodeOptions = null)
+    {
+        ...
+    }
+
+    public WidgetProxy(
+        IInvoker invoker,
+        System.Uri serviceAddressUri,
+        ProtobufEncodeOptions? encodeOptions = null)
+        : this(invoker, new ServiceAddress(serviceAddressUri), encodeOptions)
+    {
+    }
+}
+```
+
+The `invoker` parameter represents your [invocation pipeline](/icerpc/invocation/invocation-pipeline), the
+`serviceAddress` or `serviceAddressUri` parameter corresponds to the
+[address](/icerpc/invocation/service-address) of the remote service, and the `encodeOptions` parameter allows
+you to customize the encoding of your Protobuf messages. See [ProtobufEncodeOptions] for details.
+
+A `null` service address is equivalent to an icerpc service address with the default path of the associated Protobuf
+service.
+
+{% callout type="note" %}
+The default path of a Protobuf service is `/` followed by its fully qualified name. For example, the default path of
+Protobuf service `visitor_center.Greeter` is `/visitor_center.Greeter`.
+{% /callout %}
+
+### I*Name*Service
+
+Interface I*Name*Service is a server-side helper: it helps you create a service (a C# class) that implements Protobuf
+service _Name_.
+
+The principle is straightforward: your service class must be a partial class that implements I*Name*Service. It must
+also carry the [ProtobufService] attribute.
```

Comment:
Maybe use a list here to list the two things required.

Reply:
I agree, I generally like using lists in markdown. It helps everything not look like one giant block of text. I find that it also draws a bit more attention to itself.

Reply:
I'm not against a list, but I think it's also fine as-is.
There's only 2 requirements; a pretty small list.

Reply:
I think it's fine as is.